code/game/gamemodes/miniantags/pulsedemon/pulsedemon_interactions.dm

```diff
@@ -0,0 +1,275 @@
+#define PD_HIJACK_FAIL(pd) (CALLBACK(pd, TYPE_PROC_REF(/mob/living/simple_animal/demon/pulse_demon, fail_hijack)))
+
+/mob/living/simple_animal/demon/pulse_demon/ClickOn(atom/A, params)
+	if(client?.click_intercept)
+		client.click_intercept.InterceptClickOn(src, params, A)
+		return
+
+	if(next_click > world.time)
+		return
+
+	if(try_modified_click(A, params))
+		return
+
+	// escape out of whatever we've hijacked first
+	if(isapc(A) && !isturf(loc) && (A in hijacked_apcs))
+		A.attack_pulsedemon(src)
+	else if(current_weapon && istype(current_weapon, /obj/item/gun/energy))
+		// ironically, because ion guns override their emp_act, it's perfectly safe to be in one and be emp'd
+		var/obj/item/gun/energy/G = current_weapon
+		// we probably shouldn't be firing from inside a recharger or someone's bag
+		if(iscarbon(G.loc) || isturf(G.loc))
+			G.process_fire(A, src, FALSE)
+			visible_message("<span class='danger'>[G] fires itself at [A]!</span>", "<span class='danger'>You force [G] to fire at [A]!</span>", "<span class='italics'>You hear \a [G.fire_sound_text]!</span>")
+			changeNext_click(1 SECONDS) // I can't actually find what the default gun fire cooldown is, so it's 1 second until someone enlightens me
```

Comment:
time to enlighten
```suggestion
			changeNext_click(CLICK_CD_RANGE) // I can't actually find what the default gun fire cooldown is, so it's 1 second until someone enlightens me
```

Reply:
Yep I noticed this in testing 
I was like "wow this gun fires like shit" and that's because we have a much lower shooting cd